specification/mixedreality/resource-manager/Microsoft.MixedReality/preview/2019-02-28-preview/mixedreality.json

```diff
@@ -0,0 +1,724 @@
+          },
+          {
+            "$ref": "#/parameters/spatialAnchorsAccountKeyRegenerateParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SpatialAnchorsAccountKeys"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions" : {
+    "CheckNameAvailabilityRequest" : {
+      "description": "Check Name Availability Request",
+      "type": "object",
+      "required" : ["name", "type"],
+      "properties": {
+        "name": {
+          "description": "Resource Name To Verify",
+          "type": "string"
+        },
+        "type": {
+          "description": "fully qualified resource type which includes provider namespace",
+          "type": "string"
+        }
+      }
+    },
+    "CheckNameAvailabilityResponse" : {
+      "description": "Check Name Availability Response",
+      "type": "object",
+      "required" : ["nameAvailable"],
+      "properties": {
+        "nameAvailable": {
+          "description": "if name Available",
+          "$ref" : "#/definitions/NameAvailability"
+        },
+        "reason": {
+          "description": "Resource Name To Verify",
+          "$ref" : "#/definitions/NameUnavailabileReason"
+        },
+        "message": {
+          "description": "detail message",
+          "type": "string"
+        }
+      }
+    },
+    "ErrorResponse": {
+      "description": "Response on Error",
+      "type": "object",
+      "required" : ["message", "code"],
+      "properties": {
+        "message": {
+          "description": "Describes the error in detail and provides debugging information",
+          "type": "string"
+        },
+        "code": {
+          "description": "String that can be used to programmatically identify the error.",
+          "type": "string"
+        },
+        "target": {
+          "description": "The target of the particular error",
+          "type": "string"
+        },
+        "details": {
+          "description": "An array of JSON objects that MUST contain name/value pairs for code and message, and MAY contain a name/value pair for target, as described above.The contents of this section are service-defined but must adhere to the aforementioned schema.",
+          "type": "string"
+        }
+      }
+    },
+    "Operation": {
+      "description": "REST API operation",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Operation name: {provider}/{resource}/{operation}",
+          "type": "string"
+        },
+        "display": {
```

Comment:
You may consider extracting this sub-definition to its own definition named `OperationDisplay`.